integration-test-groups/cxf-soap/cxf-soap-client/src/main/java/org/apache/camel/quarkus/component/cxf/soap/client/it/CxfSoapClientRoutes.java

```diff
@@ -41,9 +41,13 @@ public class CxfSoapClientRoutes extends RouteBuilder {
     @Override
     public void configure() {
 
-        from("direct:simple")
+        from("direct:simpleUriBean")
                 .to("cxf:bean:soapClientEndpoint?dataFormat=PAYLOAD");
 
+        from("direct:simpleUriAddress")
+                .to(String.format("cxf://%s?wsdlURL=%s&dataFormat=POJO&serviceClass=%s", calculatorServiceAddress(),
+                        calculatorServiceWsdlUrl(), CalculatorService.class.getCanonicalName()));
```

Comment:
It does not make any difference for most classes incl. CalculatorService, but generally Class.getName() is the appropriate format here, because Camel is internally passing to ClassLoader.loadClass(String) which expects the Class.getName() format with `$` as a delimiter for inner classes.

```suggestion
                        calculatorServiceWsdlUrl(), CalculatorService.class.getName()));
```